upup/pkg/fi/cloudup/networking.go

```diff
@@ -106,26 +108,45 @@ const (
 	ENV_VAR_CNI_VERSION_URL = "CNI_VERSION_URL"
 )
 
-func findCNIAssets(c *api.Cluster) (string, string, error) {
+func findCNIAssets(c *api.Cluster, assetBuilder *assets.AssetBuilder) (*url.URL, string, error) {
 
 	if cniVersionURL := os.Getenv(ENV_VAR_CNI_VERSION_URL); cniVersionURL != "" {
+		u, err := url.Parse(cniVersionURL)
+		if err != nil {
+			return nil, "", nil
+		}
 		glog.Infof("Using CNI asset version %q, as set in %s", cniVersionURL, ENV_VAR_CNI_VERSION_URL)
-		return cniVersionURL, "", nil
+		return u, "", nil
 	}
 
 	sv, err := util.ParseKubernetesVersion(c.Spec.KubernetesVersion)
 	if err != nil {
-		return "", "", fmt.Errorf("Failed to lookup kubernetes version: %v", err)
+		return nil, "", fmt.Errorf("failed to lookup kubernetes version: %v", err)
 	}
 
 	sv.Pre = nil
 	sv.Build = nil
 
+	var defaultCNIAssetK8s, defaultCNIAssetHashStringK8s string
```

Comment:
Perhaps just `cniAsset` and `cniAssetHash` ?

Reply:
All this code is going away.  We have SHAs for the tar balls now :)  But I will make the change